bat-utils/lib/hapi-controllers-login.js

```diff
@@ -17,42 +15,34 @@ const v1 = {}
  */
 
 v1.login = {
-  handler: (runtime) => {
-    return async (request, reply) => {
-      if (!request.auth.isAuthenticated) return reply(boom.forbidden())
-
-      const debug = braveHapi.debug(module, request)
-      const credentials = request.auth.credentials
-      const gh = new GitHub({ debug: false })
-
-      gh.authenticate({ type: 'token', token: credentials.token })
-      gh.users.getTeams({}, (err, data) => {
-        if (err) return reply('Oops!')
+  handler: (runtime) => async (request, h) => {
+    const { auth } = request
+    const { credentials, isAuthenticated } = auth
+    if (!isAuthenticated) {
+      throw boom.forbidden()
+    }
 
-        credentials.scope = []
-        data.data.forEach(team => {
-          if (team.organization.login === runtime.login.github.organization) credentials.scope.push(team.name)
-        })
-        if (credentials.scope.length === 0) {
-          runtime.notify(debug, {
-            channel: '#devops-bot',
-            text: 'login failed ' + credentials.provider + ' ' + credentials.profile.email
-          })
-          return reply(boom.forbidden())
-        }
+    const debug = braveHapi.debug(module, request)
 
-        runtime.notify(debug, {
-          channel: '#devops-bot',
-          text: 'login ' + credentials.provider + ' ' +
-            JSON.stringify(underscore.pick(credentials.profile, [ 'username', 'displayName', 'email', 'id' ])) +
-            ': ' + JSON.stringify(credentials.scope) + ' at ' + os.hostname() + ' ' + npminfo.name + '@' +
-            npminfo.version + (process.env.DYNO ? ' at ' + process.env.DYNO : '') + ' from ' + whitelist.ipaddr(request)
-        })
-
-        request.cookieAuth.set(credentials)
-        reply.redirect(runtime.login.github.world)
+    try {
+      await checkTeams(runtime, credentials)
+    } catch (e) {
+      return runtime.notify(debug, {
```

Comment:
should we be returning `runtime.notify` or re-throwing `e`?

Reply:
rethrowing e. good catch